build/build/src/ide/web/dejavu_font.rs

```diff
@@ -0,0 +1,173 @@
+    let get_font_files = async {
+        let package = get_dejavu_font_package_(cache, octocrab).await?;
+        extract_fonts(&html_font_definitions, package, output_path).await
+    };
+    let make_css_file = async {
+        if let (Some(css_basepath), Some(css_output_path)) = (css_basepath, css_output_path) {
+            let mut css = String::new();
+            let family = "DejaVu Sans Mono";
+            for header in html_fonts {
+                use std::fmt::Write;
+                let def = html_font_definitions.get(header);
+                let def = def.ok_or_else(|| {
+                    anyhow!(
+                        "Required font not found in DejaVu Font package. \
+                  Expected a font matching: {header:?}."
+                    )
+                })?;
+                let file = def.file;
+                let weight = def.header.weight.to_number();
+                writeln!(&mut css, "@font-face {{")?;
+                writeln!(&mut css, "  font-family: '{family}';")?;
+                writeln!(&mut css, "  src: url('{css_basepath}/{file}');")?;
+                writeln!(&mut css, "  font-weight: {weight};")?;
+                writeln!(&mut css, "  font-style: normal;")?;
+                writeln!(&mut css, "}}")?;
+                writeln!(&mut css)?;
+            }
+            ide_ci::fs::tokio::write(css_output_path, css).await?;
+            Ok(())
+        } else {
+            Ok(())
+        }
+    };
+    try_join!(get_font_files, make_css_file)?;
+    Ok(())
+}
+
+pub async fn download_dejavu_sans_mono_font(
+    cache: &Cache,
+    octocrab: &Octocrab,
+    output_path: impl AsRef<Path>,
+) -> Result {
+    download_dejavu_sans_mono_font_internal(cache, octocrab, None, output_path, None::<&str>).await
+}
+
+pub async fn download_dejavu_sans_mono_font_with_css(
+    cache: &Cache,
+    octocrab: &Octocrab,
+    css_basepath: &str,
+    output_path: impl AsRef<Path>,
+    css_output_path: impl AsRef<Path>,
+) -> Result {
+    download_dejavu_sans_mono_font_internal(
+        cache,
+        octocrab,
+        Some(css_basepath),
+        output_path,
+        Some(css_output_path),
+    )
+    .await
+}
+
+/// Returns the DejaVu Font.
+pub fn dejavu_sans_mono_font() -> NonVariableDefinition {
+    DEJAVU_SANS_MONO_FONT_FAMILY_FONTS
+        .iter()
+        .map(|(name, weight)| {
+            let file = format!("{PACKAGE_FONTS_PREFIX}{PACKAGE_SANS_MONO_PREFIX}{name}.ttf");
+            let header = NonVariableFaceHeader {
+                weight: *weight,
+                width:  font::Width::Normal,
+                style:  font::Style::Normal,
+            };
+            (header, file)
+        })
+        .collect()
+}
+
+/// Extract the fonts from the given archive file, and write them in the given directory.
+#[context("Failed to extract fonts from archive {}", package.as_ref().display())]
+pub async fn extract_fonts(
```

Comment:
The code seems copied from `lib/rust/enso-font/src/lib.rs`, just the archive format is different?
(should be documented and likely placed in a shared location, this does not seem deja-vu-specific)

Reply:
there are also slight differences in archive structure - but in general, yeah. I'll see whether I can make it more general.

Reply:
i *can* make it more general... but not sure about placing it in a shared location. specifically:
- i think it doesn't belong in `ide_ci`, even though `enso-enso-font` *does* depend on `ide_ci`
- i'd prefer avoiding adding another crate in `lib/rust/`
- i'd imagine `enso-enso-font` is going away soon anyway, once GUI1 is removed. at that point it would be able to be fully moved into `src/ide/web` next to the other fonts.

Reply:
`ide_ci` crate name is archaic, conceptually build script have:
* `ide_ci` for general-purpose build-time-needed utilities; (so not only CI and not only IDE)
* `enso_build` for enso-specific logic;
* `enso_build_cli` as a CLI driver for the former.